src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java

```diff
@@ -198,266 +195,243 @@ public WardPharmacyRectify(JFrame owner, Ward ward, Medical medical) {
+		gbcPanelLot.gridwidth = 2;
+		contentPanel.add(getJPanelLot(), gbcPanelLot);
+
+		jLabelInLot = new JLabel(MessageBundle.getMessage("angal.medicalstockward.rectify.inlot")); //$NON-NLS-1$
+		jLabelInLot.setVisible(false);
+		GridBagConstraints gbcLabelInLot = new GridBagConstraints();
+		gbcLabelInLot.anchor = GridBagConstraints.SOUTH;
+		gbcLabelInLot.insets = new Insets(0, 0, 5, 0);
+		gbcLabelInLot.gridx = 4;
+		gbcLabelInLot.gridy = 3;
+		contentPanel.add(jLabelInLot, gbcLabelInLot);
+
+		JLabel jLabelNewQuantity = new JLabel(MessageBundle.getMessage("angal.medicalstockward.rectify.actualquantity")); //$NON-NLS-1$
+		GridBagConstraints gbcLabelNewQuantity = new GridBagConstraints();
+		gbcLabelNewQuantity.anchor = GridBagConstraints.EAST;
+		gbcLabelNewQuantity.insets = new Insets(0, 0, 5, 5);
+		gbcLabelNewQuantity.gridx = 0;
+		gbcLabelNewQuantity.gridy = 4;
+		contentPanel.add(jLabelNewQuantity, gbcLabelNewQuantity);
+
+		GridBagConstraints gbcSpinnerNewQty = new GridBagConstraints();
+		gbcSpinnerNewQty.fill = GridBagConstraints.HORIZONTAL;
+		gbcSpinnerNewQty.insets = new Insets(0, 0, 5, 5);
+		gbcSpinnerNewQty.gridx = 1;
+		gbcSpinnerNewQty.gridy = 4;
+		gbcSpinnerNewQty.gridwidth = 2;
+		contentPanel.add(getJSpinnerNewQty(), gbcSpinnerNewQty);
+
+		GridBagConstraints gbcLabelLotQty = new GridBagConstraints();
+		gbcLabelLotQty.insets = new Insets(0, 0, 5, 0);
+		gbcLabelLotQty.gridx = 4;
+		gbcLabelLotQty.gridy = 4;
+		contentPanel.add(getJLabelLotQty(), gbcLabelLotQty);
+
+		JLabel jLabelReason = new JLabel(MessageBundle.getMessage("angal.medicalstockward.rectify.reason")); //$NON-NLS-1$
+		GridBagConstraints gbcLabelReason = new GridBagConstraints();
+		gbcLabelReason.anchor = GridBagConstraints.EAST;
+		gbcLabelReason.insets = new Insets(0, 0, 0, 5);
+		gbcLabelReason.gridx = 0;
+		gbcLabelReason.gridy = 5;
+		contentPanel.add(jLabelReason, gbcLabelReason);
+
+		JTextField jTextFieldReason = new JTextField();
+		GridBagConstraints gbcTextFieldReason = new GridBagConstraints();
+		gbcTextFieldReason.insets = new Insets(0, 0, 0, 5);
+		gbcTextFieldReason.fill = GridBagConstraints.HORIZONTAL;
+		gbcTextFieldReason.gridx = 1;
+		gbcTextFieldReason.gridy = 5;
+		gbcTextFieldReason.gridwidth = 2;
+		contentPanel.add(jTextFieldReason, gbcTextFieldReason);
+
+		/*
+		 * TODO: to refactor all this part by extracting in separated method all this logic
+		 */
+		JPanel jButtonPanel = new JPanel();
+		jButtonPanel.setLayout(new FlowLayout(FlowLayout.CENTER));
+		getContentPane().add(jButtonPanel, BorderLayout.SOUTH);
+
+		JButton jButtonOk = new JButton(MessageBundle.getMessage("angal.common.ok.btn"));
+		jButtonOk.setMnemonic(MessageBundle.getMnemonic("angal.common.ok.btn.key"));
+		jButtonOk.addActionListener(actionEvent -> {
+			Object item;
+			Medical med;
+
 			/*
-			 * TODO: to refactor all this part by extracting in separated method all this logic
+			 *  To override MovWardBrowserManager.validateMovementWard() behavior
```

Comment:
Why to remove the TODO reminder?

Reply:
because it is redundant; see line 329

Reply:
ok, but you also changed the comment, so they are 2 different TODOs to take care?

Reply:
I put the comment back